src/libmongoc/tests/test-mongoc-versioned-api.c

```diff
@@ -135,6 +135,43 @@ _test_mongoc_server_api_client_pool (void)
    mongoc_uri_destroy (uri);
 }
 
+static void
+_test_mongoc_server_api_client_pool_once (void)
+{
+   mongoc_uri_t *uri;
+   mongoc_client_pool_t *pool;
+   mongoc_client_t *client;
+   mongoc_server_api_t *api;
+   bson_error_t error;
+
+   uri = mongoc_uri_new ("mongodb://localhost");
+   pool = mongoc_client_pool_new (uri);
+
+   api = mongoc_server_api_new (MONGOC_SERVER_API_V1);
+
+   client = mongoc_client_pool_pop (pool);
+   BSON_ASSERT (!client->api);
+
+   /* Cannot change server API once a client has been popped. */
+   ASSERT (!mongoc_client_pool_set_server_api (pool, api, &error));
+   ASSERT_ERROR_CONTAINS (
+      error,
+      MONGOC_ERROR_POOL,
+      MONGOC_ERROR_POOL_API_TOO_LATE,
+      "Cannot set server api after a client has been created");
+
+   ASSERT (!mongoc_client_set_server_api (client, api, &error));
+   ASSERT_ERROR_CONTAINS (error,
+                          MONGOC_ERROR_CLIENT,
+                          MONGOC_ERROR_CLIENT_API_ALREADY_SET,
+                          "Cannot set server api more than once");
```

Comment:
This is interesting.  So, when we get the client from the pool, it does *not* have an api set (`client->api` is NULL) but we've set the `client->api_set` flag to prevent users from setting the `api` directly on the client?  It's a bit of a misleading message, I would rather that the message said something like "can't set server api directly on a pooled client, please set it on the parent pool"

Reply:
As per above, I'll change the flag name to `is_pooled` and will check that for pooled client. This makes the flag reusable in CDRIVER-3537 if we decide to block setting other properties.